tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Session.java

```diff
@@ -373,6 +413,15 @@ public Runner setOptions(RunOptions options) {
       return this;
     }
 
+    private void checkInitialization() {
```

Comment:
This check worries me. Initializers aren't persisted in the GraphDef, and so if you deserialize a graph you can't find them. In Tribuo we have to run two sets of initializers to get around this problem, one for the graph that the user supplied, and one for the gradient, loss and outputs that we add to the graph before training.

I don't think this check is going to cause issues right now because it's not persisted, but we need to think carefully about how it interacts with persistence.